lib/sneakers/worker.rb

```diff
@@ -92,6 +92,16 @@ def do_work(delivery_info, metadata, msg, handler)
 
         metrics.increment("work.#{self.class.name}.ended")
       end #process
+    rescue => ex
+      # If the thread pool is shutting down, it will throw a RuntimeError
+      # So, requeue the message as it is not processed.
+      handler.reject(delivery_info, metadata, msg, true)
+
+      # If is was not the thread pool shutting down, let's know about the error:
+      unless ( ex.to_s =~ /unable to add work while shutting down/i )
```

Comment:
It would be really nice to avoid depending on a particular exception message here. If we need to make Bunny throw a custom exception type, I'm all for it.